File: src/main/java/tn/esprit/com/foyer/entities/Chambre.java
package tn.esprit.com.foyer.entities;

import jakarta.persistence.*;
import lombok.*;

import java.io.Serializable;
import java.util.Set;

@Entity
@Getter
@Setter
@Builder

@AllArgsConstructor
@NoArgsConstructor
@Table( name = "Chambre")
public class Chambre implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "idChambre")
    private Long idChambre; // Clé primaire
    private Long numeroChambre;
    @Enumerated(EnumType.STRING)
    private TypeChambre typeC;
    @ManyToOne
    Bloc bloc;
    @OneToMany(cascade = CascadeType.ALL ,fetch = FetchType.EAGER)
    private Set<Reservation> reservations;




}

File: src/app/classes/Chambre.ts
// chambre.model.ts
 //Assurez-vous d'importer correctement l'enum TypeChambre
import { Bloc } from './Bloc';
import { TypeChambre } from './TypeChambre';
export class Chambre {
  idChambre: number;
  numeroChambre: number;
  typeC: TypeChambre;
  //reservations: Reservation[]; // Assurez-vous d'importer correctement la classe Reservation
  bloc: Bloc; // Assurez-vous d'importer correctement la classe Bloc

  constructor(
    idChambre: number,
    numeroChambre: number,
    typeC: TypeChambre,
    //reservations: Reservation[],
    //bloc: Bloc
  ) {
    this.idChambre = idChambre;
    this.numeroChambre = numeroChambre;
    this.typeC = typeC;
    //this.reservations = reservations;
    //his.bloc = bloc;
  }
}
